dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumWebViewClient.kt

```diff
@@ -0,0 +1,121 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-2020 Datadog, Inc.
+ */
+
+package com.datadog.android.rum
+
+import android.graphics.Bitmap
+import android.net.http.SslError
+import android.os.Build
+import android.webkit.SslErrorHandler
+import android.webkit.WebResourceError
+import android.webkit.WebResourceRequest
+import android.webkit.WebResourceResponse
+import android.webkit.WebView
+import android.webkit.WebViewClient
+import androidx.annotation.RequiresApi
+import com.datadog.android.rum.internal.domain.RumEventSerializer
+
+/**
+ * A [WebViewClient] propagating all relevant events to the [GlobalRum] monitor.
+ *
+ * This will map the page loading, and webview errors into Rum Resource and
+ * Error events respectively.
+ */
+open class RumWebViewClient : WebViewClient() {
+
+    // region WebViewClient
+
+    /** @inheritdoc */
+    override fun onPageStarted(view: WebView?, url: String?, favicon: Bitmap?) {
+        super.onPageStarted(view, url, favicon)
+        if (url != null) {
+            GlobalRum.get().startResource(
+                url,
+                url
+            )
+        }
+    }
+
+    /** @inheritdoc */
+    override fun onPageFinished(view: WebView?, url: String?) {
+        super.onPageFinished(view, url)
+        if (url != null) {
+            GlobalRum.get().stopResource(
+                url,
+                RumResourceKind.DOCUMENT
+            )
+        }
+    }
+
+    /** @inheritdoc */
+    override fun onReceivedError(
+        view: WebView?,
+        errorCode: Int,
+        description: String?,
+        failingUrl: String?
+    ) {
+        super.onReceivedError(view, errorCode, description, failingUrl)
+        GlobalRum.get().addError(
+            "Error $errorCode: $description",
+            ORIGIN,
+            null,
+            mapOf(RumEventSerializer.TAG_HTTP_URL to failingUrl)
```

Comment:
Hmm not a big fan of adding this key from the RumEventSerializer. I would rather add a new constant here in this class and remove don't mix the RumEventSerializer here. It feels like this doesn't belong here.

Reply:
The thing is that this is a standard key. Duplicating the constant makes it more error prone (if we change the field in RumEventSerializer we might forget to change it here or vice versa). 
The goal of a constant is to be constant everywhere.

Reply:
Still...this tells me that if I want to use the RumMonitor API (by passing some attributes) I need to know the internal classes (e.g. RumEventSerializer) and its internal tags. Also what will happen if for example you want to extract this in another module (artifact). You will not be able to do that without exposing the RumEventSerializer class as this is an internal. Maybe instead we should add new method inside the RumMonitor -> `addError(message, origin, failingUrl, attributes)`

Reply:
This is our own implementation, so we do know the internals. People using the RUM monitor can use whatever attributes they want, they won't look into this.
If I need to extract it into a different module, then I'll duplicate the constant.

No let's not add this new method. Errors don't necessarily have to do with urls, so this is a specific use case.

Reply:
Nevermind...I will accept it but still not very convinced.